### PR TITLE
[Merged by Bors] - Simulator and attestation service fixes

### PR DESCRIPTION
**Pull request by pawanjay176**
## Issue Addressed

#1729 #1730 

Which issue # does this PR address?

## Proposed Changes

1. Fixes a bug in the simulator where nodes can't find each other due to 0 udp ports in their enr.
2. Fixes bugs in attestation service where we are unsubscribing from a subnet prematurely.

More testing is needed for attestation service fixes.

### CODE REVIEW

**Comment by AgeManning** [comment]
Does this still need further testing? 

Also clippy

**Comment by pawanjay176** [comment]
@AgeManning I think the simulator works fine now. Haven't seen any peering issues. 
Trying to fix some randomly failing tests in the attestation service. The tests failed when the subnet for the short term subscription was same as the long term random subscription so we ended up with unexpected number of events. 

Added another test to check that we don't unsubscribe early from subnets when 2 validator subscriptions have the same subnet id but different slots.

File: beacon_node/network/src/attestation_service/tests/mod.rs [hunk 1]
```diff
@@ -20,7 +20,7 @@ mod tests {
     use tempfile::tempdir;
     use types::{CommitteeIndex, EthSpec, MinimalEthSpec};
 
-    const SLOT_DURATION_MILLIS: u64 = 200;
+    const SLOT_DURATION_MILLIS: u64 = 400;
```
Comment (pawanjay176) [inline beacon_node/network/src/attestation_service/tests/mod.rs:23]: @AgeManning had to increase this to make the `test_same_subnet_unsubscription` pass. 

In the test, we are trying to create 2 validator subscriptions with the same `subnet_id` but different slots. 
The issue is that `slots_per_epoch` for `MinimalEthSpec` is just 8. So `compute_subnet_id` gives different results when `current_slot`(which is random) is >= 7. By increasing the `SLOT_DURATION`, i'm hoping `current_slot` will never go greater than that. Keen to see if there's a better way.

Comment (AgeManning) [reply beacon_node/network/src/attestation_service/tests/mod.rs:23]: Yeah some of these tests probably need to be re-worked. 
Happy for a hack fix at the moment

**Review by AgeManning — APPROVED** [review]
bors r+

**Pull request closed without merging**